docs/configuration.md

```diff
@@ -0,0 +1,473 @@
+>##
+>>### config.playback.options
+>>##### Type: `PKPlaybackOptionsObject`
+>>```js
+>>{
+>>  html5: {
+>>    hls: Object,
+>>    dash: Object
+>>  }
+>>}
+>>```
+>>##### Default: 
+>>```js
+>>{
+>>  html5: {
+>>    hls: {},
+>>    dash: {}
+>>  }
+>>}
+>>```
+>>##### Description: Defines the media source adapters configurations.
+>>* For `hls` configuration, see the [hls.js](https://github.com/video-dev/hls.js/blob/master/doc/API.md#fine-tuning) documentation.
+>>* For `dash` configuration, see the [shaka-player](https://shaka-player-demo.appspot.com/docs/api/tutorial-config.html) documentation.
+>## 
+>>### config.playback.preferNative
+>>##### Type: `PKPreferNativeConfig`
+>>```js
+>>{
+>>  hls: boolean,
+>>  dash: boolean
+>>}
+>>```
+>>##### Default: 
+>>```js
+>>{
+>>  hls: false,
+>>  dash: false
+>>}
+>>```
+>>##### Description: Indicates whether to prefer native browser playback (if supported) with media source extensions.
+>>If one of the values is set to True and the player chooses to play the truthly media source extension, the player will try to play it natively if supported by the browser. 
+>> #### Example:
+>>Lets assume the following configuration:
+>>```js
+>>var config = {
+>>  playback: {
+>>    preferNative:  {
+>>      hls: true
+>>    }
+>>  }
+>>};
+>>```
+>> If the player is running on a _Safari_ browser, the player will use the native hls playback managed by the _Safari_ browser. However, if running on a browser in which hls playback is not supported natively, for example, _Chrome_, the player will play hls using the `hls.js` library.
+>## 
+>>### config.playback.streamPriority
+>>##### Type: `Array<PKStreamPriorityObject`
+>>>##### Type `PKStreamPriorityObject`
+>>>```js
+>>>{
+>>>   engine: string,
+>>>   format: string
+>>>}
+>>>```
+>>##### Default: 
+>>```js
+>>[
+>>   {
+>>     engine: "html5",
+>>     format: "hls"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "dash"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "progressive"
+>>    }
+>>]
+>>```
+>>##### Description: Specifies the list of engine and stream format pairs of the player by ascending order.
+>>As soon as the player receives the sources, it will review the configuration array and try to play the source with the matched stream format according to the matched engine. 
+>>For example, in the priority configuration above, the player will try to play the hls stream using an html5 engine first. If an hls stream isn't received, the player will continue to play the dash stream using an html5 engine. If a dash stream isn't received, the player will then will continue to play the progressive stream using an html5 engine.
+
+### Workflow between `sources`, `preferNative` & `streamPriority`
```

Comment:
VERY IMPORTANT 👍 better separate to different doc - Source Selection Logic
@AnnaAleksandrowicz - FYI

Reply:
OK - So a new page with a link from here? Perhaps the title should be "How does the Source Selection Logic Work?"

Reply:
@AnnaAleksandrowicz sounds good to me

Reply:
@dan-ziv  - can you create the new page for the source selection with the content that's currently in this configuration page?

Reply:
Sure